src/app/utils/dom-utils.ts

```diff
@@ -229,3 +229,35 @@ export function getCommonAncestor(node1, node2) {
 export function createNsResolver(doc: Document) {
   return (prefix: string) => prefix === 'ns' ? doc.documentElement.namespaceURI : undefined;
 }
+
+export function updateCSS(rules: Array<[string, string]>) {
+  const thisCSS = document.styleSheets[0];
+  rules.forEach((rule) => {
+    thisCSS.insertRule(`${rule[0]} {${rule[1]}}`, 0);
+  });
+}
+
+export function deepSearch(obj, attrToMatch: string, valuesToMatch: any[], limit: number = 4000) {
+  const results = [];
+  let counter = limit;
+  function search(obj) {
+    for (const key in obj) {
+      const value = obj[key];
+      if ((key === attrToMatch) && (valuesToMatch.includes(obj[attrToMatch]))) {
+        results.push(obj);
+      }
+      if (typeof value === 'object' && value !== null) {
+        if (counter > 0) {
+          search(value);
+          counter = counter - 1;
+        } else {
+          console.log('EVT WARN: element is too deep, not searching further in', obj);
+        }
+      }
+    }
+  }
+  search(obj);
+
+  return results;
```

Comment:
Can you find a better way? Defining a function within a function is not that good. Maybe you can pass the counter as a param of the search and define search outside